### ISSUE TITLE
Timeout config fails

### ISSUE THREAD

**Issue by Nimamoh** [kotest/kotest#879]
Hello,

Using the release `3.3.2`. This example test runs 10 seconds then success

```kotlin
class TimeoutFails : StringSpec() {

    init {
        "Timeout fails".config(timeout = Duration.ofSeconds(1L)) {
            assertAll(10, Gen.string()) {
                Thread.sleep(1000)
            }
        }
    }
}
```

According to the doc, I'd expect the test to be cancelled after 1 second and fail

**Comment by sksamuel** [comment]
It should yes. I'll investigate.

**Comment by sksamuel** [comment]
Fixed in 3.4 and I'll back port to 3.3.4
